src/AzureExtension/DevBox/DevBoxProvider.cs

```diff
@@ -135,8 +135,23 @@ public IAsyncOperation<ComputeSystemsResult> GetComputeSystemsAsync(IDeveloperId
             }
             catch (Exception ex)
             {
-                Log.Logger()?.ReportError($"Unable to get all Dev Boxes", ex);
-                return new ComputeSystemsResult(ex, Resources.GetResource(Constants.DevBoxUnableToPerformOperationKey, ex.Message), ex.Message);
+                var errorMessage = string.Empty;
+                if (ex.InnerException != null && ex.InnerException.Message.Contains("Account has previously been signed out of this application"))
+                {
+                    errorMessage = Resources.GetResource(Constants.RetrivalFailKey) + Resources.GetResource(Constants.SessionExpiredKey);
+                }
+                else if (ex.Message.Contains("A passthrough token was detected without proper resource provider context"))
+                {
+                    errorMessage = Resources.GetResource(Constants.RetrivalFailKey) + Resources.GetResource(Constants.UnconfiguredKey);
+                }
+                else
+                {
+                    errorMessage = Resources.GetResource(Constants.RetrivalFailKey) + ex.Message;
+                    return new ComputeSystemsResult(ex, Resources.GetResource(Constants.RetrivalFailKey), ex.Message);
+                }

```

Comment:
Interestingly do you know if the exception message is guaranteed to be in english?

Reply:
No, we will just display it in the language it is sent by the API.

Reply:
spoke off line, we aren't sure if the exception message is ever localized, but worse case scenario, we display the full error message